smartcosmos-framework/src/main/java/net/smartcosmos/events/rest/RestSmartCosmosEventTemplate.java

```diff
@@ -62,10 +62,13 @@ public void convertAndSend(SmartCosmosEvent<Object> message)
         eventHttpHeaders.set("Authorization", String.format("%s %s", tokenType, accessToken.getValue()));
 
         try {
-            smartCosmosEventTaskExecutor.execute(() -> restOperations.exchange(eventUri,
```

Comment:
This diff points me to the hopefully right place for the fix: The `restOperations` instance still has the old `OAuth2ClientContext`, but creating a new one based on the token in the `RequestContextHolder` should do the trick.

To make that work, we only need to override the `DispatcherListener` and add a `RequestContextFilter`, both with thread inheritance enabled.